### ISSUE TITLE
regular spikes in loss (cats-and-dogs notebook)

### ISSUE THREAD

**Issue by markusloecher** [fchollet/deep-learning-with-python-notebooks#191]
I am struggling with explaining the regular spikes in both training and validation losses observed when running  the cats-and-dogs CNN from the book, see image below:

![image](https://user-images.githubusercontent.com/7353291/145383575-6fc00e62-c05f-4404-b0b3-a899bff230eb.png)

There is an interesting discussion of a  very similar issue on SO:

[Why does my training loss have regular spikes?](https://stackoverflow.com/questions/47824598/why-does-my-training-loss-have-regular-spikes)

, but my data size is an integer multiple of the batch size, so the explanation does not hold for me. I am unable to explain the sharp spikes and the steep drops in training loss at the end/beginning of each epoch.

I am happy to provide the notebook I used.

Thx!

**Comment by ifond** [comment]
YES!!! I have received your E-mail——Steven Lee
